### ISSUE TITLE
Uncorect comp error

### ISSUE THREAD

**Issue by MatejMagat305** [vlang/v#17262]
### Describe the bug

vab compiling didn't succeed.
vab.android.CompileError: failed to compile .c to .o:
vab.android.compile_v_imports_c_dependencies: getting NDK sysroot path.
vab.android.ndk.sysroot_path couldn't locate sysroot at "/data/data/com.termux/files/home/android-ndk-r23c/toolchains/llvm/prebuilt//sysroot". You could try with a newer NDK version (>= r19).

It caused by https://github.com/vlang/vab/blob/685820f97c1afa79ccbca87715581228b1bde98e/android/ndk/ndk.v#L221

### Expected Behavior

Funtion return one of string

### Current Behavior

Return empty string
It probably related with #17258, because after this was crash

### Reproduction Steps

Well I don't know whether is meaningfull to termux, but probably it is problem of whole vab

### V version

V 0.3.3 47e8d96

### Environment details (OS name and version, etc.)

OS: termux, 4.14.190-perf-25421105-abP619XXU1BVK6, #1 SMP PREEMPT Fri Nov 25 19:21:57 KST 2022
Processor: 8 cpus, 64bit, little endian
CC version: clang version 15.0.7

getwd: /data/data/com.termux/files/home
vmodules: /data/data/com.termux/files/home/.vmodules
vroot: /data/data/com.termux/files/home/v
vexe: /data/data/com.termux/files/home/v/v
vexe mtime: 2023-02-08 20:27:15
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.3 fbfdab9.47e8d96

Git version: git version 2.39.1
Git vroot status: weekly.2023.06-9-g47e8d967
.git/config present: true
thirdparty/tcc status: thirdparty-unknown-unknown de82a130

**Comment by felipensp** [comment]
What error message is shown? Can you write a short reproducible case?

**Comment by MatejMagat305** [comment]
> What error message is shown? Can you write a short reproducible case?

och sorry, it is not exciplit enough ..., this is error on termux from vab:
```
vab compiling didn't succeed.
vab.android.CompileError: failed to compile .c to .o:
vab.android.compile_v_imports_c_dependencies: getting NDK sysroot path.
vab.android.ndk.sysroot_path couldn't locate sysroot at
/data/data/com.termux/files/home/android-ndk-r23c/toolchains/llvm/prebuilt//sysroot
```
I notice that after `prebuilt/` and before `/sysroot` is empty space, which is related contrete funkcion ..., and how to reproduce?
so install termux and termux and try compile:
```
module main
pub fn host_arch() string {
	$if linux {
		return 'linux-x86_64'
	} $else $if macos {
		return 'darwin-x86_64'
	} $else $if windows {
		return 'windows-x86_64'
	} $else $if termux {
		return 'linux-aarch64'
	} $else {
		return 'unknown'
	}
}
fn main(){
       println(host_arch())
}
``` 
and see print empty line

**Comment by JalonSolov** [comment]
That's really strange... even if none of the `$if`s match, it should return `unknown`, not `''`.

**Comment by MatejMagat305** [comment]
> That's really strange... even if none of the `$if`s match, it should return `unknown`, not `''`.

it is litle bit similar to #17038, but this is more evident ... .
Therefore it is strange I put there like error ...

**Comment by MatejMagat305** [comment]
but it is weird before I was compile right way at making video ...

**Comment by felipensp** [comment]
Try changing your code for:

```V
	$if termux {
		return 'linux-aarch64'
	} $else $if linux {
		return 'linux-x86_64'
	} $else $if macos {
		return 'darwin-x86_64'
	} $else $if windows {
		return 'windows-x86_64'
	} $else {
		return 'unknown'
	}```

**Comment by MatejMagat305** [comment]
> Try changing the code to:
> 
> ```v
> 	$if termux {
> 		return 'linux-aarch64'
> 	} $else $if linux {
> 		return 'linux-x86_64'
> 	} $else $if macos {
> 		return 'darwin-x86_64'
> 	} $else $if windows {
> 		return 'windows-x86_64'
> 	} $else {
> 		return 'unknown'
> 	}
> ```

Yes, it is working on termux, but will it working on other or is termux sometime go tu linux branch?

**Comment by MatejMagat305** [comment]
When I try this: 
```
pub fn host_arch() string {
        $if linux {
                return 'linux-x86_64'
        }
        $if macos {
                return 'darwin-x86_64'
        }
        $if windows {
                return 'windows-x86_64'
        }
        $if termux {
                return 'linux-aarch64'
        }
        return 'unknown'
}```
It is working too

**Comment by Larpon** [comment]
I wonder if this is why I initially used the runtime [os.user_os()](https://github.com/vlang/v/blob/master/vlib/os/os.v#L358) instead, note how it's similarly structured without the use of `$else` 🤔
